webapp/src/test/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationServiceTest.java

```diff
@@ -233,4 +252,108 @@ void translateBranch(Branch branch) throws ExecutionException, InterruptedExcept
     // make sure the stats are ready for whatever is done next in notification
     branchStatisticService.computeAndSaveBranchStatistics(branch);
   }
+
+  @Test
+  public void safeTranslationNotification() throws Exception {
```

Comment:
This test currently seems to cover several different scenarios, e.g.:
tracked branch + commit + new safe translated message, tracked branch + commit + 10 hour delay etc.

Could each individual scenario be split into its own test - to make it easier to follow the logic and also ensure that when something does break - we know exactly which scenarios still work and which ones are broken?